exporter.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import time
+from os import getenv
+
+import os
+
+import urllib3
+from prometheus_client import REGISTRY, start_http_server
+
+from collectors.PyVimServiceCollector import PyVimServiceCollector
+from collectors.SshServiceCollector import SshServiceCollector
+from collectors.EsxiOverallStateCollector import EsxiOnlineStateCollector
+from modules.Exceptions import VCenterException
+
+# init logger once
+logger = logger = logging.getLogger('esxi-exporter')
+logger.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+formatter = logging.Formatter('[%(asctime)s] [%(levelname)s] %(message)s')
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+# disable ssl warning
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+class Exporter:
+
+    def run_prometheus_server(self, port: int) -> None:
+        """
+        Starts a prometheus server which will stay running
+        :param port: The port the server should listen on
+        """
+
+        # startup prometheus server
+        logger.info("starting http server...")
+        start_http_server(port)
+
+        # register collectors
+        if not bool(getenv('DISABLE_PYVIM', False)):
+            logger.info("registering collector: PyVimServiceCollector... ")
+            REGISTRY.register(PyVimServiceCollector())
+        if not bool(getenv('DISABLE_SSH', False)):
+            logger.info("registering collector: sshServiceCollector...")
+            REGISTRY.register(SshServiceCollector())
+        if not bool(getenv('DISABLE_OVERALLSTATE', False)):
+            logger.info("registering collector: EsxiOverallStateCollector...")
+            REGISTRY.register(EsxiOnlineStateCollector())

```

Comment:
As already mentioned, setup in configfile and implement a factory pattern to register and import on demand.
Ref:
https://github.com/sapcc/vrops-exporter/blob/master/exporter.py#L92-L105

Reply:
collectors can now be registered with `config.yaml`:

```yaml
collectors:
# As the classname
  - EsxiOverallStateCollector
  - SshServiceCollector
```